### PR TITLE
Adding a new service to request/get geofence

### PR DESCRIPTION
**Pull request by haarshitgarg**
Trying to fix issue #25715

Will be adding more geofence messages to this pull request

### CODE REVIEW

File: Tools/ros2/ardupilot_msgs/srv/GeoFence.srv [hunk 1]
```diff
@@ -0,0 +1,13 @@
+
+# This service requests the vehicle to arm or disarm its motors.
```
Comment (Ryanf55) [inline Tools/ros2/ardupilot_msgs/srv/GeoFence.srv:2]: Can you fix all the comments that talk about arming?

File: libraries/AP_DDS/AP_DDS_Client.cpp [hunk 2]
```diff
@@ -633,6 +635,45 @@ void AP_DDS_Client::on_request(uxrSession* uxr_session, uxrObjectId object_id, u
...
+        ardupilot_msgs_srv_GeoFence_Response geofence_data;
+
+        const bool deserialize_success = ardupilot_msgs_srv_GeoFence_Request_deserialize_topic(ub, &geofence_request);
+        if (deserialize_success == false) {
+            break;
+        }
+        
+        if(geofence_request.req)
+        {
+            geofence_data.radius = AP::fence()->get_radius();
```
Comment (Ryanf55) [inline libraries/AP_DDS/AP_DDS_Client.cpp:649]: While ardupilot supports both fence types of `AC_FENCE_TYPE_CIRCLE` and `AC_FENCE_TYPE_POLYGON`, this message population seems to assume it's a circle. 

Also, what is the plan for the `PolyFence` support in ArduPilot, which has inclusion/exclusions.

**Review by Ryanf55 — COMMENTED** [review]
Thanks for the contribution. This is good start! I'd like to better understand how you see this working in a ROS system - does ArduPilot only store fences and then the ROS computer can read the current fences? What about if the ROS computer wanted to set fences?

**Comment by haarshitgarg** [comment]
> Thanks for the contribution. This is good start! I'd like to better understand how you see this working in a ROS system - does ArduPilot only store fences and then the ROS computer can read the current fences? What about if the ROS computer wanted to set fences?

Hello, sorry for the inconsistencies. I will push some changes with better framed messages. Right now I was only working on a getter but yes setter could be implemented as well. I am testing the new message structure that is why I kept this as draft. Will soon put them here for review. Thanks a lot :)

**Comment by haarshitgarg** [comment]
I wasn't able to work on it so I am closing this draft PR

**Pull request closed without merging**